tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/ServerlessMiniAgent.cs

```diff
@@ -0,0 +1,185 @@
+    {
+        var isGCPFunction = GetIsGCPFunction();
+        var isAzureFunction = GetIsAzureFunction();
+        if (!isGCPFunction && !isAzureFunction)
+        {
+            return null;
+        }
+
+        Log.Information("Trying to start the mini agent");
+
+        string rustBinaryPath;
+        if (Environment.GetEnvironmentVariable("DD_MINI_AGENT_PATH") != null)
+        {
+            rustBinaryPath = Environment.GetEnvironmentVariable("DD_MINI_AGENT_PATH");
+        }
+        else
+        {
+            // Environment.OSVersion.Platform can return PlatformID.Unix on MacOS, this is OK as GCP & Azure don't have MacOs functions.
+            if (Environment.OSVersion.Platform != PlatformID.Unix && Environment.OSVersion.Platform != PlatformID.Win32NT)
+            {
+                Log.Error("Serverless Mini Agent is only supported on Windows and Linux.");
+                return null;
+            }
+
+            var isWindows = os == PlatformID.Win32NT;
+            var dirPathSep = isWindows ? "\\" : "/";
+
+            string rustBinaryPathRoot;
+            if (isGCPFunction)
+            {
+                rustBinaryPathRoot = "/layers/google.dotnet.publish/publish/bin";
+            }
+            else
+            {
+                // IsAzureFunction
+                if (isWindows)
+                {
+                    rustBinaryPathRoot = "C:\\home\\site\\wwwroot";
+                }
+                else
+                {
+                    // linux
+                    rustBinaryPathRoot = "/home/site/wwwroot";
+                }
+            }
+
+            string rustBinaryPathOsFolder = isWindows ? "datadog-serverless-agent-windows-amd64" : "datadog-serverless-agent-linux-amd64";
+            string rustBinaryFileExtension = isWindows ? ".exe" : string.Empty;
+            rustBinaryPath = string.Format("{0}{1}{2}{3}datadog-serverless-trace-mini-agent{4}", rustBinaryPathRoot, dirPathSep, rustBinaryPathOsFolder, dirPathSep, rustBinaryFileExtension);
+        }
+
+        return rustBinaryPath;
+    }
+
+    internal static bool GetIsGCPFunction()
+    {
+        bool isDeprecatedGCPFunction = Environment.GetEnvironmentVariable(ConfigurationKeys.GCPFunction.DeprecatedFunctionNameKey) != null && Environment.GetEnvironmentVariable(ConfigurationKeys.GCPFunction.DeprecatedProjectKey) != null;
+        bool isNewerGCPFunction = Environment.GetEnvironmentVariable(ConfigurationKeys.GCPFunction.FunctionNameKey) != null && Environment.GetEnvironmentVariable(ConfigurationKeys.GCPFunction.FunctionTargetKey) != null;
+
+        return isDeprecatedGCPFunction || isNewerGCPFunction;
+    }
+
+    internal static bool GetIsAzureFunction()
+    {
+        return Environment.GetEnvironmentVariable(ConfigurationKeys.AzureAppService.FunctionsExtensionVersionKey) != null && Environment.GetEnvironmentVariable(ConfigurationKeys.AzureAppService.SiteNameKey) != null;
+    }
+
+    internal static string GetGCPAzureFunctionName()
```

Comment:
Ideally, there should be a speration of concern. All Serverless helper methods should be IMO in ImmutableSettings (even if they are static). This class should contain only the mini agent related stuff (ie `StartServerlessMiniAgent`, `MiniAgentDataReceivedHandler`, `ProcessMiniAgentLog`, `GetMiniAgentPath`).

Reply:
done, separated out the two GetIsXFunction methods into the immutable settings files.